### PR TITLE
Iryna kazlova/user reports about mistake

### PR DESCRIPTION
**Pull request by rinchan603**
Add implementation commands and query for reports

### CODE REVIEW

File: Elevel/Application/Features/ReportCommands/CreateReportCommand.cs [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+
+        public class Validator : AbstractValidator<CreateReportCommand.Request>
+        {
+            public Validator()
+            {
+                this.CascadeMode = CascadeMode.Stop;
+                
+                When(x => x.QuestionId.HasValue, () =>
+                {
+                    RuleFor(x => x.QuestionId).NotEmpty().NotEqual(Guid.Empty)
```
Comment (parys) [inline Elevel/Application/Features/ReportCommands/CreateReportCommand.cs:36]: here should be AuditionId ?

File: Elevel/Application/Features/ReportCommands/GetReportDetailQuery.cs [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+            {
+                _context = context;
+                _mapper = mapper;
+                _userManager = userManager;
+            }
+
+            public async Task<Response> Handle(Request request, CancellationToken cancellationToken)
+            {
+                var report = await _context.Reports
+                    .FirstOrDefaultAsync(a => a.Id == request.Id, cancellationToken);
```
Comment (parys) [inline Elevel/Application/Features/ReportCommands/GetReportDetailQuery.cs:38]: or mapper here or AsNoTracking

File: Elevel/Elevel.Api/Controllers/ReportController.cs [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using System;
+using System.Threading.Tasks;
+using Elevel.Application.Features.ReportCommands;
+using Elevel.Domain.Enums;
+
+namespace Elevel.Api.Controllers
+{
+    [Authorize, Route("api/[controller]")]
```
Comment (parys) [inline Elevel/Elevel.Api/Controllers/ReportController.cs:12]: any role can reports?

Comment (rinchan603) [reply Elevel/Elevel.Api/Controllers/ReportController.cs:12]: Yes, any role.

File: Elevel/Application/Features/ReportCommands/GetReportDetailQuery.cs (hunk 2 above)
Comment (rinchan603) [reply Elevel/Application/Features/ReportCommands/GetReportDetailQuery.cs:38]: Done

File: Elevel/Application/Features/ReportCommands/CreateReportCommand.cs (hunk 1 above)
Comment (rinchan603) [reply Elevel/Application/Features/ReportCommands/CreateReportCommand.cs:36]: Add AuditionId

**Review by parys — APPROVED** [review]

**Review by DShcharbakou — APPROVED** [review]

File: Elevel/Domain/Enums/Status.cs [hunk 4]
```diff
@@ -0,0 +1,10 @@
+﻿namespace Elevel.Domain.Enums
+{
+    public enum ReportStatus : byte
+    {
+        Unknown = 0,
```
Comment (andreyrogov8) [inline Elevel/Domain/Enums/Status.cs:5]: Why it is needed?

**Pull request merged**